src/Raven.Server/Json/BlittableJsonTextWriterExtensions.cs

```diff
@@ -271,6 +272,137 @@ public static void WriteEtlTaskProgress<TWriter>(this TWriter writer, JsonOperat
             writer.WriteEndObject();
         }
 
+        public static void WriteReplicationTaskProgress<TWriter>(this TWriter writer, JsonOperationContext context, IEnumerable<IReplicationTaskProgress> progress)
+        where TWriter : IBlittableJsonTextWriter
+        {
+            writer.WriteStartObject();
```

Comment:
I wonder if all this can be done by implementing `ToJson()` on `IReplicationTaskProgress` or `taskStats.ProcessesProgress`?

Similar to `EtlHandlerProcessorForDebugStats`